src/Server.php

```diff
@@ -57,13 +63,13 @@ public function handleRequest(ConnectionInterface $conn, Request $request, $body
         $response = new Response($conn);
         $response->on('close', array($request, 'close'));
 
-        if (!$this->listeners('request')) {
-            $response->end();
-
-            return;
+        $callback = $this->callback;
+        try {
+            $callback($request, $response);
+        } catch (\Exception $ex) {
+            $this->emit('error', array());
```

Comment:
Not sure this makes sense here, can you give some usage examples? How could I possibly react to when my callback throws an exception? Can I trigger a clean shutdown or end the response etc.?

(It looks like this may be better off in a separate PR?)